airflow/models/taskinstance.py

```diff
@@ -1052,6 +1098,14 @@ def check_and_change_state_before_execution(
         :return: whether the state was changed to running or not
         :rtype: bool
         """
+        if mark_success is not None:
+            warnings.warn(
+                'The argument `mark_success` has been deprecated. Use `state=State.SUCCESS` instead.',
+                DeprecationWarning,
```

Comment:
```suggestion
                DeprecationWarning,
                stacklevel=3
```
(+1 for the `@provide_session`)